opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -532,14 +569,17 @@ public boolean batchInsertEvents(JSONArray array, long serverVersion) {
         return batchInsertEvents(array, serverVersion, getWritableDatabase());
     }
 
-    public boolean batchInsertEvents(JSONArray array, long serverVersion, SQLiteDatabase sqLiteDatabase) {
+    public boolean batchInsertEvents(JSONArray array, long serverVersion, SQLiteDatabase
+            sqLiteDatabase) {
         if (array == null || array.length() == 0) {
             return false;
         }
 
         SQLiteStatement insertStatement = null;
         SQLiteStatement updateStatement = null;
 
+        populateFormSubmissionIds();
```

Comment:
this should be checking the form submission ids  from the events in the JSONArray

Reply:
What if there are other events already saved?

Reply:
this is to check the new events if they are already saved on the database. What other events are you referring to